spec/services/publish_draft_edition_service_spec.rb

```diff
@@ -16,12 +16,18 @@
         publish_request = stub_publishing_api_publish(edition.content_id,
                                                       update_type: nil,
                                                       locale: edition.locale)
+        time = nil
+        freeze_time do
+          time = Time.current
+          described_class.call(edition, user, with_review: true)
+        end
 
-        described_class.call(edition, user, with_review: true)
         expect(publish_request).to have_been_requested
         expect(edition.document.live_edition).to eq(edition)
         expect(edition).to be_published
         expect(edition).to be_live
+        expect(edition.published_at).to eq(time)
+        expect(edition.document.first_published_at).to eq(time)
```

Comment:
Sorry to have not brought this up in the last review but I don't think this is the right place for this test as we've got a lot of expectations here and the "it" doesn't really capture this.

I'd suggest you have a separate it for example:

```
it "sets the document first publishing time" do
  freeze_time do
    expect { described_class.call(document.current_edition, user, with_review: true) }
      .to change(document, :first_published_at).to(Time.current)
  end
end
```

Then it's probably worth adding a similar expectation outside of any contexts (like `it "calls the PublishAssetsService"` for instance) which asserts that edition has a published_at timestamp set to current time.

Reply:
Sure, I've added a separate test which tests the editions published_at timestamp instead of piggy backing on another test!